File: genshin-cost-calculator/scripts/download-images.py
import json
import os
import urllib.request
from pathlib import Path

# 데이터 파일 로드
with open('public/data/characters.json', 'r', encoding='utf-8') as f:
    characters_data = json.load(f)

with open('public/data/weapons.json', 'r', encoding='utf-8') as f:
    weapons_data = json.load(f)

# 이미지 폴더 생성
os.makedirs('public/images/characters', exist_ok=True)
os.makedirs('public/images/weapons', exist_ok=True)

# 캐릭터 이미지 다운로드
print("캐릭터 이미지 다운로드 중...")
for character in characters_data['characters']:
    char_id = character['id']
    char_name = character['name']
    
    # 캐릭터 ID에서 URL용 이름 생성 (첫 글자만 대문자)
    url_name = ''.join(word.capitalize() for word in char_id.split('_'))
    image_url = f"https://gi.yatta.moe/assets/UI/UI_AvatarIcon_{url_name}.png"
    file_path = f"public/images/characters/{char_id}.png"
    
    try:
        if not os.path.exists(file_path):
            urllib.request.urlretrieve(image_url, file_path)
            print(f"✓ {char_name} ({char_id})")
        else:
            print(f"⊘ {char_name} (이미 존재함)")
    except Exception as e:
        print(f"✗ {char_name}: {str(e)}")

# 무기 이미지 다운로드
print("\n무기 이미지 다운로드 중...")
for weapon in weapons_data['weapons']:
    weapon_id = weapon['id']
    weapon_name = weapon['name']
    
    # 무기 ID에서 URL용 이름 생성
    url_name = '_'.join(part.capitalize() for part in weapon_id.replace('sword_', '').replace('claymore_', '').replace('polearm_', '').replace('catalyst_', '').replace('bow_', '').split('_'))
    
    # 무기 타입 추가
    if weapon_id.startswith('sword_'):
        weapon_type = 'Sword'
        base_name = weapon_id.replace('sword_', '')
    elif weapon_id.startswith('claymore_'):
        weapon_type = 'Claymore'
        base_name = weapon_id.replace('claymore_', '')
    elif weapon_id.startswith('polearm_'):
        weapon_type = 'Pole'
        base_name = weapon_id.replace('polearm_', '')
    elif weapon_id.startswith('catalyst_'):
        weapon_type = 'Catalyst'
        base_name = weapon_id.replace('catalyst_', '')
    else:  # bow
        weapon_type = 'Bow'
        base_name = weapon_id.replace('bow_', '')
    
    # URL 이름 생성 (camelCase → CamelCase)
    url_name = ''.join(word.capitalize() for word in base_name.split('_'))
    
    # gi.yatta.moe URL 패턴 추측 (정확하지 않을 수 있음)
    image_url = f"https://gi.yatta.moe/assets/UI/UI_EquipIcon_{weapon_type}_{url_name}.png"
    file_path = f"public/images/weapons/{weapon_id}.png"
    
    try:
        if not os.path.exists(file_path):
            urllib.request.urlretrieve(image_url, file_path)
            print(f"✓ {weapon_name} ({weapon_id})")
        else:
            print(f"⊘ {weapon_name} (이미 존재함)")
    except Exception as e:
        print(f"✗ {weapon_name}: {str(e)}")

print("\n✅ 다운로드 완료!")

File: genshin-cost-calculator/scripts/download-images.js
const fs = require('fs');
const path = require('path');
const https = require('https');

// 이미지를 다운로드하고 저장하는 함수
function downloadImage(url, filePath) {
  return new Promise((resolve, reject) => {
    const dir = path.dirname(filePath);
    
    // 디렉토리가 없으면 생성
    if (!fs.existsSync(dir)) {
      fs.mkdirSync(dir, { recursive: true });
    }

    const file = fs.createWriteStream(filePath);
    https.get(url, (response) => {
      response.pipe(file);
      file.on('finish', () => {
        file.close();
        resolve();
      });
    }).on('error', (err) => {
      fs.unlink(filePath, () => {}); // 실패 시 파일 삭제
      reject(err);
    });
  });
}

// 캐릭터 이미지 다운로드
async function downloadCharacterImages() {
  const charactersData = JSON.parse(fs.readFileSync(path.join(__dirname, '../public/data/characters.json'), 'utf8'));
  
  console.log('캐릭터 이미지 다운로드 중...');
  
  for (const character of charactersData.characters) {
    const fileName = character.id + '.png';
    const filePath = path.join(__dirname, '../public/images/characters', fileName);
    
    // 이미지 URL에서 캐릭터 ID 추출
    const urlParts = character.icon.split('_');
    const characterName = urlParts[urlParts.length - 1].replace('.png', '');
    const imageUrl = `https://gi.yatta.moe/assets/UI/UI_AvatarIcon_${characterName}.png`;
    
    try {
      await downloadImage(imageUrl, filePath);
      console.log(`✓ ${character.name} (${fileName})`);
    } catch (err) {
      console.error(`✗ ${character.name}: ${err.message}`);
    }
  }
  
  console.log('캐릭터 이미지 다운로드 완료!');
}

// 무기 이미지 다운로드
async function downloadWeaponImages() {
  const weaponsData = JSON.parse(fs.readFileSync(path.join(__dirname, '../public/data/weapons.json'), 'utf8'));
  
  console.log('무기 이미지 다운로드 중...');
  
  for (const weapon of weaponsData.weapons) {
    const fileName = weapon.id + '.png';
    const filePath = path.join(__dirname, '../public/images/weapons', fileName);
    
    // 무기 이미지 URL 생성
    const urlMatch = weapon.icon.match(/UI_EquipIcon_(.+?)\.png/);
    if (urlMatch) {
      const imageUrl = `https://gi.yatta.moe/assets/UI/UI_EquipIcon_${urlMatch[1]}.png`;
      
      try {
        await downloadImage(imageUrl, filePath);
        console.log(`✓ ${weapon.name} (${fileName})`);
      } catch (err) {
        console.error(`✗ ${weapon.name}: ${err.message}`);
      }
    }
  }
  
  console.log('무기 이미지 다운로드 완료!');
}

// 메인 함수
async function main() {
  try {
    await downloadCharacterImages();
    await downloadWeaponImages();
    console.log('\n✅ 모든 이미지 다운로드 완료!');
  } catch (err) {
    console.error('에러:', err);
    process.exit(1);
  }
}

main();
